### PR TITLE
[diffusion] model: support FLUX.2-klein-base

### PR DESCRIPTION
**Pull request by alex0dd**
## Motivation

At the moment, Flux.2-Klein-Base model is indirectly supported within SGLang via distilled Flux.2-Klein implementation, which however doesn't allow negative prompts and CFG parallelism.
This PR solves the issue by adding Flux.2-Klein-Base configs and fixes required to make its negative prompts work.

## Modifications

* Added FLUX.2-klein-base pipeline config:
   * New `Flux2KleinBasePipelineConfig`, extending `Flux2KleinPipelineConfig` with `should_use_guidance=True` (Klein-base is undistilled and uses guidance embeddings).
   * Implemented `prepare_neg_cond_kwargs` to build `freqs_cis` for negative-prompt embeddings on the CFG path.
* Added FLUX.2-klein-base sampling defaults:
   * New `Flux2KleinBaseSamplingParams`, extending `FluxSamplingParams` with `num_inference_steps=50`, `guidance_scale=4.0`, and an empty default `negative_prompt`.
* Added FLUX.2-klein-base to multimodal registry:
   * Updated the existing Klein detector to exclude `"base"` so distilled and undistilled variants are routed to their respective configs.
* Generalized empty-string negative-prompt handling:
   * In `python/sglang/multimodal_gen/runtime/pipelines_core/stages/text_encoding.py`, relaxed the CFG `negative_prompt` validator to accept `""` (`isinstance(x, str)`) so Klein-base's empty default doesn't trip validation.
* Added 1-GPU diffusion CI test case for FLUX.2-klein-base text-to-image and the measured perf baseline.

## Accuracy Tests

N/A for this PR.
This change mainly integrates a new model/pipeline into the framework and does not modify existing model forward logic or kernel behavior.

## Speed Tests and Profiling

N/A for this PR.
No speed regression/performance optimization is claimed in this change.

## Checklist

- [x] Format your code according to the [Format code with pre-commit](https://docs.sglang.io/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [x] Add unit tests according to the [Run and add unit tests](https://docs.sglang.io/developer_guide/contribution_guide.html#run-and-add-unit-tests).
- [x] Update documentation according to [Write documentations](https://docs.sglang.io/developer_guide/contribution_guide.html#write-documentations).
- [x] Provide accuracy and speed benchmark results according to [Test the accuracy](https://docs.sglang.io/developer_guide/contribution_guide.html#test-the-accuracy) and [Benchmark the speed](https://docs.sglang.io/developer_guide/contribution_guide.html#benchmark-the-speed).
- [x] Follow the SGLang code style [guidance](https://docs.sglang.io/developer_guide/contribution_guide.html#code-style-guidance).

## Review and Merge Process

1. Ping Merge Oncalls to start the process. See the [PR Merge Process](https://github.com/sgl-project/sglang/blob/main/.github/MAINTAINER.md#pull-request-merge-process).
2. Get approvals from [CODEOWNERS](https://github.com/sgl-project/sglang/blob/main/.github/CODEOWNERS) and other reviewers.
3. Trigger CI tests with [comments](https://docs.sglang.io/developer_guide/contribution_guide.html#how-to-trigger-ci-tests) or contact authorized users to do so.
   - Common commands include `/tag-and-rerun-ci`, `/tag-run-ci-label`, `/rerun-failed-ci`
4. After green CI and required approvals, ask Merge Oncalls or people with Write permission to merge the PR.

---
### CI States

Latest PR Test (Base): :white_check_mark: [Run #26262056974](https://github.com/sgl-project/sglang/actions/runs/26262056974)
Latest PR Test (Extra): :x: [Run #26262056860](https://github.com/sgl-project/sglang/actions/runs/26262056860)

### CODE REVIEW

**Comment by mickqian** [comment]
could you attach some results and compare it with official output? cheers

**Comment by alex0dd** [comment]
> could you attach some results and compare it with official output? cheers

Sure, attached the results and generation code below.

Text-to-image:
| prompt | diffusers | this PR |
| --- | --- | --- |
| A macro wildlife photo of a green frog in a rainforest pond, highly detailed, eye-level shot | klein_base_t2i_0 | klein_base_t2i_0 |
| A bedroom with large windows and modern furniture, gray and gold, luxurious, mid century modern style, modern city | klein_base_t2i_1 | klein_base_t2i_1 |


Edit:

| prompt | input image | diffusers | this PR |
| --- | --- | --- | --- |
| Make the frog purple. | klein_base_t2i_0 | klein_base_edit_0 | klein_base_edit_0 |
| Make it snow outside the window. | klein_base_t2i_1 | klein_base_edit_1 | klein_base_edit_1 |

Code used to generate them:
<details>

<summary>SGLang (text-to-image and edit)</summary>

```bash
set -euo pipefail
output_dir=/output/klein_base_compare/sglang
mkdir -p "$output_dir"

T2I_PROMPTS=(
  "A macro wildlife photo of a green frog in a rainforest pond, highly detailed, eye-level shot"
  "A bedroom with large windows and modern furniture, gray and gold, luxurious, mid century modern style, modern city"
)
EDIT_PROMPTS=(
  "Make the frog purple."
  "Make it snow outside the window."
)

# Pass 1: T2I -> klein_base_t2i_{0,1}.png
for i in 0 1; do
  echo ">>> sglang T2I [$i]: ${T2I_PROMPTS[$i]}"
  sglang generate \
    --model-path black-forest-labs/FLUX.2-klein-base-4B \
    --num-gpus 1 --sp-degree 1 \
    --prompt "${T2I_PROMPTS[$i]}" \
    --negative-prompt "" \
    --width 1024 --height 1024 \
    --output-path "$output_dir" \
    --output-file-name "klein_base_t2i_${i}" \
    --save-output \
    --seed 0 --guidance-scale 4.0 --num-inference-steps 50
done

# Pass 2: T2I -> edit
for i in 0 1; do
  echo ">>> sglang edit [$i]: ${EDIT_PROMPTS[$i]}"
  sglang generate \
    --model-path black-forest-labs/FLUX.2-klein-base-4B \
    --num-gpus 1 --sp-degree 1 \
    --prompt "${EDIT_PROMPTS[$i]}" \
    --negative-prompt "" \
    --image-path "${output_dir}/klein_base_t2i_${i}.png" \
    --output-path "$output_dir" \
    --output-file-name "klein_base_edit_${i}" \
    --save-output \
    --seed 0 --guidance-scale 4.0 --num-inference-steps 50
done

echo
echo "Done. Outputs in $output_dir:"
ls -la "$output_dir"
```

</details>

<details>

<summary>Diffusers (text-to-image and edit, adapted from <a href="https://huggingface.co/black-forest-labs/FLUX.2-klein-base-4B#using-with-diffusers-%F0%9F%A7%A8">HF page</a>)</summary>

```python
import os
import torch
from PIL import Image
from diffusers import Flux2KleinPipeline

OUT_DIR = "/output/klein_base_compare/diffusers"
os.makedirs(OUT_DIR, exist_ok=True)

T2I_PROMPTS = [
    "A macro wildlife photo of a green frog in a rainforest pond, highly detailed, eye-level shot",
    "A bedroom with large windows and modern furniture, gray and gold, luxurious, mid century modern style, modern city",
]
EDIT_PROMPTS = [
    "Make the frog purple.",
    "Make it snow outside the window.",
]

pipe = Flux2KleinPipeline.from_pretrained(
    "black-forest-labs/FLUX.2-klein-base-4B", torch_dtype=torch.bfloat16
).to("cuda")


def generate(prompt, image=None):
    return pipe(
        prompt=prompt,
        image=image,
        height=1024,
        width=1024,
        num_inference_steps=50,
        guidance_scale=4.0,
        generator=torch.Generator(device="cuda").manual_seed(0),
    ).images[0]


for i, prompt in enumerate(T2I_PROMPTS):
    print(f"T2I [{i}] {prompt}")
    generate(prompt).save(f"{OUT_DIR}/klein_base_t2i_{i}.png")

for i, prompt in enumerate(EDIT_PROMPTS):
    print(f"edit [{i}] {prompt}")
    init = Image.open(f"{OUT_DIR}/klein_base_t2i_{i}.png").convert("RGB")
    generate(prompt, image=init).save(f"{OUT_DIR}/klein_base_edit_{i}.png")
```

</details>

**Review by mickqian — APPROVED** [review]

**Comment by alex0dd** [comment]
@mickqian any next steps on this? I see that some tests are failing, but they're not related to this PR (rather from main branch updates)

**Comment by mickqian** [comment]
@alex0dd fantastic job, cheers
